public/scripts/model/repo.js

```diff
@@ -0,0 +1,18 @@
+'use strict';
+
+(function(module) {
+  const repos = {};
+
+  repos.all = [];
+
+  repos.requestRepos = function(callback) {
```

Comment:
You may need to refactor this into an `.ajax` request, rather than a `.get` request, because you're having async issues.